src/process/windows/SDL_windowsprocess.c

```diff
@@ -0,0 +1,452 @@
+#include "SDL_internal.h"
+
+#ifdef SDL_PROCESS_WINDOWS
+
+#include "../../core/windows/SDL_windows.h"
+#include "../SDL_sysprocess.h"
+#include "../../file/SDL_iostream_c.h"
+
+#define READ_END 0
+#define WRITE_END 1
+
+struct SDL_ProcessData {
+    PROCESS_INFORMATION process_information;
+};
+
+static void CleanupStream(void *userdata, void *value)
+{
+    SDL_Process *process = (SDL_Process *)value;
+    const char *property = (const char *)userdata;
+
+    SDL_ClearProperty(process->props, property);
+}
+
+static bool SetupStream(SDL_Process *process, HANDLE handle, const char *mode, const char *property)
+{
+    SDL_IOStream *io = SDL_IOFromHandle(handle, mode, true);
+    if (!io) {
+        return false;
+    }
+
+    SDL_SetPointerPropertyWithCleanup(SDL_GetIOProperties(io), "SDL.internal.process", process, CleanupStream, (void *)property);
+    SDL_SetPointerProperty(process->props, property, io);
+    return true;
+}
+
+static bool SetupRedirect(SDL_PropertiesID props, const char *property, HANDLE *result)
+{
+    SDL_IOStream *io = (SDL_IOStream *)SDL_GetPointerProperty(props, property, NULL);
+    if (!io) {
+        SDL_SetError("%s is not set", property);
+        return false;
+    }
+
+    HANDLE handle = (HANDLE)SDL_GetPointerProperty(SDL_GetIOProperties(io), SDL_PROP_IOSTREAM_WINDOWS_HANDLE_POINTER, INVALID_HANDLE_VALUE);
+    if (handle == INVALID_HANDLE_VALUE) {
+        SDL_SetError("%s doesn't have SDL_PROP_IOSTREAM_WINDOWS_HANDLE_POINTER available", property);
+        return false;
+    }
+
+    if (!DuplicateHandle(GetCurrentProcess(), handle,
+                         GetCurrentProcess(), result,
+                         0, TRUE, DUPLICATE_SAME_ACCESS)) {
+        WIN_SetError("DuplicateHandle()");
+        return false;
+    }
+    return true;
+}
+
+static bool join_arguments(const char * const *args, char **args_out)
+{
+    size_t len;
+    int i;
+    int i_out;
+    char *result;
+
+    len = 0;
+    for (i = 0; args[i]; i++) {
+        const char *a = args[i];
+
+        /* two double quotes to surround an argument with */
+        len += 2;
+
+        for (; *a; a++) {
+            switch (*a) {
+            case '"':
+                len += 2;
+                break;
+            default:
+                len += 1;
+                break;
+            }
+        }
+        /* space separator or final '\0' */
+        len += 1;
+    }
+
+    result = SDL_malloc(len);
+    if (!result) {
+        *args_out = NULL;
+        return false;
+    }
+
+    i_out = 0;
+    for (i = 0; args[i]; i++) {
+        const char *a = args[i];
+
+        result[i_out++] = '"';
+        for (; *a; a++) {
+            switch (*a) {
+            case '"':
+                result[i_out++] = '\\';
+                result[i_out++] = *a;
+                break;
+            default:
+                result[i_out++] = *a;
+                break;
+            }
+        }
+        result[i_out++] = '"';
+        result[i_out++] = ' ';
```

Comment:
Has this been tested? IIRC, on Windows, double quotes are escaped by doubling the double quotes, like:

`a "b" c` -> `"a ""b"" c"`

Also, backslashes are not escaped; should they be?

Reply:
Double quotes are tested in `process_testArguments` in `testprocess`.
It looks like there is no test for backslashes.

Reply:
Good question. Want to add a test for this?

Reply:
This should be tested in process_testArguments(), can you see if there's anything not covered?